src/components/ProbeEditor.tsx

```diff
@@ -14,40 +14,40 @@ import {
   InputControl,
 } from '@grafana/ui';
 import { Label as SMLabel, Probe, OrgRole } from 'types';
-import { SMDataSource } from 'datasource/DataSource';
 import { hasRole } from 'utils';
 import SMLabelsForm from 'components/SMLabelsForm';
 import ProbeStatus from './ProbeStatus';
 import { validateLabel } from 'validation';
+import { InstanceContext } from 'components/InstanceContext';
 
 interface Props {
   probe: Probe;
-  instance: SMDataSource;
   onReturn: (reload: boolean) => void;
 }
 
 const minInputWidth = css`
   min-width: 200px;
 `;
 
-const ProbeEditor: FC<Props> = ({ probe, instance, onReturn }) => {
+const ProbeEditor: FC<Props> = ({ probe, onReturn }) => {
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const [showTokenModal, setShowTokenModal] = useState(false);
   const [probeToken, setProbeToken] = useState('');
+  const { instance } = useContext(InstanceContext);
 
   const onSave = async (formValues: Probe) => {
     // Form values always come back as a string, even for number inputs
     formValues.latitude = Number(formValues.latitude);
     formValues.longitude = Number(formValues.longitude);
 
     if (probe.id) {
-      await instance.updateProbe({
+      await instance?.api.updateProbe({
         ...probe,
         ...formValues,
       });
       onReturn(true);
     } else {
-      const info = await instance.addProbe({
+      const info = await instance?.api.addProbe({
```

Comment:
`instance?`  

maybe short circuit early if instance is null, or somehow make it not optional?